### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Example: Fixes #1234. See also #3456.
+Please use keywords (e.g., Fixes) to create link to the issues or pull requests
+you resolved, so that they will automatically be closed when your pull request
+is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
+-->
+
+
+#### What does this implement/fix? Explain your changes.
+
+#### How should this PR be tested?
```
Comment (LennartPurucker) [inline .github/PULL_REQUEST_TEMPLATE.md:23]:
```suggestion
#### Minimal Example / How should this PR be tested?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+<!--
+This only needs to be filled if there is anything to test.
```
Comment (LennartPurucker) [inline .github/PULL_REQUEST_TEMPLATE.md:26]:
```suggestion
Please provide a minimal example / test that showcases running the code with your changes. 
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+<!--
+Please be aware that we are a loose team of volunteers so patience is
+necessary; assistance handling other issues is very welcome. We value
+all user contributions, no matter how minor they are. If we are slow to
+review, either the pull request needs some benchmarking, tinkering,
+convincing, etc. or more likely the reviewers are simply busy. In either
+case, we ask for your understanding during the review process.
+
+Thanks for contributing!
+-->
```
Comment (LennartPurucker) [inline .github/PULL_REQUEST_TEMPLATE.md:41]:
```suggestion
-->

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
```

Comment (LennartPurucker) [reply .github/PULL_REQUEST_TEMPLATE.md:41]: I copied this from AutoGluon, it might be good. Feel free to reject.

**Review by eddiebergman — APPROVED** [review]

**Comment by LennartPurucker** [comment]
@eddiebergman I think your merge was a bit too quick for me 😅

**Review by LennartPurucker — COMMENTED** [review]
LGTM, made some minor suggestions that I would prefere when writing PRs.

**Comment by LennartPurucker** [comment]
We can keep the current template and ignore my suggestions above. It was only minor stuff.

**Comment by eddiebergman** [comment]
I'll add a new PR with those changes, apologies for being a bit fast with it

**Pull request merged**